features/telemetry/protocols/netflow/parser/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/ParserBase.java

```diff
@@ -241,7 +265,32 @@ public void setThreads(int threads) {
                     // if we can't keep up
                     final Runnable dispatch = () -> {
                         // Let's serialize
-                        byte[] flowMessage = buildMessage(record, enrichment);
+                        byte[] flowMessage = new byte[0];
+                        try {
+                            flowMessage = buildMessage(record, enrichment);
+                        } catch (IllegalFlowException e) {
+                            final Optional<Instant> instant = illegalFlowEventCache.getUnchecked(remoteAddress.getAddress());
+
+                            if (!instant.isPresent() || Duration.between(instant.get(), Instant.now()).getSeconds() > getIllegalFlowEventRate()) {
+                                illegalFlowEventCache.put(remoteAddress.getAddress(), Optional.of(Instant.now()));
+
+                                eventForwarder.sendNow(new EventBuilder()
+                                        .setUei(ILLEGAL_FLOW_EVENT_UEI)
+                                        .setTime(new Date())
+                                        .setSource(getName())
+                                        .setInterface(remoteAddress.getAddress())
+                                        .setDistPoller(identity.getId())
+                                        .addParam("monitoringSystemId", identity.getId())
+                                        .addParam("monitoringSystemLocation", identity.getLocation())
+                                        .setParam("cause", e.getMessage())
+                                        .setParam("illegalFlowEventRate", (int) getIllegalFlowEventRate())
+                                        .getEvent());
+                            }
+
+                            LOG.warn("Illegal flow detected", e);
```

Comment:
Can we use a rate limited logger or move this up into the block above? (Trying to avoid spamming the logs with this.)

Can we also include the exporter address in the log message?

Reply:
Pas de problème. La ligne a été déplacée vers le bloc supérieur.

Reply:
C'est très bien @christianpape. Je suis ravi.